iothub/device/src/Authentication/IIotHubConnectionInfo.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Azure.Devices.Client.Transport;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Interface for device configurations and common attributes.
+    /// </summary>
+    internal interface IIotHubConnectionInfo : IAuthorizationProvider
```

Comment:
Moq can only mock interfaces and virtual methods, so this interface is meant to replace the IDeviceIdentity interface.

Reply:
I'd prefer `IDeviceIdentity` over this name. Thoughts?

Reply:
I like IDeviceIdentity for this interface